### DISCUSSION TITLE
DnnlExecutionProvider missing?

### DISCUSSION THREAD

**Discussion by dzhao** [pykeio/ort#2] [General]
Hi, first of all great crate!! much cleaner and more comprehensive.
I am ready to use it in one of my critical project. 
One thing thought as we are using CPU for serving and our CPUs will benefit greatly from mkl lib.
Onnx has DnnlExecutionProvider that will enable the mkl support but it doesn't seem your crate provide this support..
Also I checked the downloaded onnx lib in my Mac and I don't see the dnnl lib so it looks like it is just using the normal MLAS cpu ep.
Is it possible to add this option? 
Thanks so much for the work!

**Comment by decahedron1** [comment]
Tracking this in #3. I'm sick at the moment so this may take a day or two to implement, sorry.

**Comment by decahedron1** [comment]
[`ort` v1.13.2](https://crates.io/crates/ort/1.13.2) has support for `ExecutionProvider::dnnl()` among a few other fixes. You'll need to compile ONNX Runtime from source with oneDNN support and point `ort` to the libraries using [the `system` strategy](https://github.com/pykeio/ort#strategies) (also make sure you enable the `onednn` Cargo feature).

**Comment by dzhao** [comment]
great fix. thanks!
while you are at it maybe also add openvino? This is yet another cpu ep.
Not urgent. I can probably add a pr if you prefer.

**Reply by decahedron1** [reply]
I'm currently looking into DirectML support, so it may take a while for me to get to OpenVINO. I would greatly appreciate a PR though.
